config/configManager.go

```diff
@@ -0,0 +1,186 @@
+limitations under the License.
+*/
+
+package config
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"sync"
+	"time"
+
+	ef "github.com/deislabs/ratify/pkg/executor/core"
+	"github.com/deislabs/ratify/pkg/policyprovider"
+	pf "github.com/deislabs/ratify/pkg/policyprovider/factory"
+	"github.com/deislabs/ratify/pkg/referrerstore"
+	sf "github.com/deislabs/ratify/pkg/referrerstore/factory"
+	"github.com/deislabs/ratify/pkg/verifier"
+	vf "github.com/deislabs/ratify/pkg/verifier/factory"
+	"github.com/fsnotify/fsnotify"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+var (
+	configHash string
+)
+
+// Create a executor from configurationFile and setup config file watcher
+func GetExecutorAndWatchForUpdate(configFilePath string) (*ef.Executor, error) {
+	cf, err := Load(configFilePath)
+
+	if err != nil {
+		return &ef.Executor{}, err
+	}
+
+	configHash = cf.FileHash
+
+	stores, verifiers, policyEnforcer, err := createFromConfig(cf)
+
+	if err != nil {
+		return &ef.Executor{}, err
+	}
+
+	executor := ef.Executor{
+		Verifiers:      verifiers,
+		ReferrerStores: stores,
+		PolicyEnforcer: policyEnforcer,
+		Config:         &cf.ExecutorConfig,
+		Mu:             sync.RWMutex{},
+	}
+
+	err = watchForConfigurationChange(configFilePath, &executor)
+
+	if err != nil {
+		return &ef.Executor{}, err
+	}
+
+	logrus.Info("configuration successfully loaded.")
+
+	return &executor, nil
+}
+
+// Returns created referer store, verifier, policyprovider objects from config
+func createFromConfig(cf Config) ([]referrerstore.ReferrerStore, []verifier.ReferenceVerifier, policyprovider.PolicyProvider, error) {
+	stores, err := sf.CreateStoresFromConfig(cf.StoresConfig, GetDefaultPluginPath())
+
+	if err != nil {
+		return nil, nil, nil, errors.Wrap(err, "failed to load store from config")
+	}
+	logrus.Infof("stores successfully created. number of stores %d", len(stores))
+
+	verifiers, err := vf.CreateVerifiersFromConfig(cf.VerifiersConfig, GetDefaultPluginPath())
+
+	if err != nil {
+		return nil, nil, nil, errors.Wrap(err, "failed to load verifiers from config")
+	}
+
+	logrus.Infof("verifiers address %v ,%v ", verifiers[0].Name(), &verifiers[0])
+	logrus.Infof("verifiers successfully created. number of verifiers %d", len(verifiers))
+
+	policyEnforcer, err := pf.CreatePolicyProviderFromConfig(cf.PoliciesConfig)
+
+	if err != nil {
+		return nil, nil, nil, errors.Wrap(err, "failed to load policy provider from config")
+	}
+
+	logrus.Infof("policies successfully created.")
+
+	return stores, verifiers, policyEnforcer, nil
+}
+
+// Setup a watcher on file at configFilePath, reload executor on file change
+func watchForConfigurationChange(configFilePath string, executor *ef.Executor) error {
+
+	watcher, err := fsnotify.NewWatcher()
+	if err != nil {
+		errors.Wrap(err, "new file watcher on configuration file failed ")
+	}
+
+	go func() {
+		for {
+			logrus.StandardLogger().Infof("see this msg every 30sec %v ", configFilePath)
```

Comment:
this is for debugging the configMap update in a cluster environment. This will be removed